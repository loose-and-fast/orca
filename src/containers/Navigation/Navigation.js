import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';
import { Menu, Grid, Sticky } from 'semantic-ui-react';
import Intro from '../../components/Intro/Intro.js';
import Portfolio from '../Portfolio/Portfolio.js';
import Contact from '../Contact/Contact.js';
import Bio from '../../components/Bio/Bio.js';
import './Navigation.css';

class Navigation extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activeItem: 'closest'
    };
    this.handleItemClick = this.handleItemClick.bind(this);
  }

  handleItemClick(e, { name }) {
    this.setState({ activeItem: name });
  }

  render() {
    return (
      <Grid columns={3} id="clue">
        <Router>
          <Grid.Row centered>
            <Grid.Column width={3}>
              <Sticky>
                <Menu text vertical size="massive" inverted className="nav1">
                  <Link to="/">
                    <Menu.Item
                      active={activeItem === 'closest'}
                      onClick={this.handleItemClick}
                    >
                      home
                    </Menu.Item>
                  </Link>
                  {/* <Link to='/portfolio'><Menu.Item  active={activeItem === 'portfolio'} onClick={this.handleItemClick}>portfolio</Menu.Item></Link> */}
                  <Link to="/bio">
                    <Menu.Item
                      active={activeItem === 'bio'}
                      onClick={this.handleItemClick}
                    >
                      about us
                    </Menu.Item>
                  </Link>
                  <Link to="/contact">
                    <Menu.Item
                      active={activeItem === 'contact'}
                      onClick={this.handleItemClick}
                    >
                      contact
                    </Menu.Item>
                  </Link>
                </Menu>
              </Sticky>
            </Grid.Column>

            <Grid.Column width={10} textAlign="middle">
              <Route exact path="/" component={Intro} />
              {/* <Route path="/portfolio" component={Portfolio}/> */}
              <Route path="/bio" component={Bio} />
              <Route path="/contact" component={Contact} />
            </Grid.Column>

            <Grid.Column width={3} />
          </Grid.Row>
        </Router>
      </Grid>
    );
  }
}

Navigation.propTypes = {};

export default Navigation;
