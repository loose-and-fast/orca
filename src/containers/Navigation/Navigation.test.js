import React from 'react'
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';

import Navigation from './Navigation'

describe('Navigation', () => {

  it('should match the snapshot', () => {
    const nav = renderer.create(
      <StaticRouter context={{}} >
        <Navigation />
      </StaticRouter>
    ).toJSON();
    expect(nav).toMatchSnapshot()
  })
})