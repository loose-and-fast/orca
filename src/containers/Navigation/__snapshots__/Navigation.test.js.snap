// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match the snapshot 1`] = `
<div
  className="ui three column grid"
  id="clue"
>
  <div
    className="centered row"
  >
    <div
      className="three wide column"
    >
      <div
        className={undefined}
      >
        <div />
        <div
          style={Object {}}
        >
          <div
            className="ui massive inverted text vertical nav1 menu"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <a
                className="active item"
                onClick={[Function]}
              >
                home
              </a>
            </a>
            <a
              href="/bio"
              onClick={[Function]}
            >
              <a
                className="item"
                onClick={[Function]}
              >
                about us
              </a>
            </a>
            <a
              href="/contact"
              onClick={[Function]}
            >
              <a
                className="item"
                onClick={[Function]}
              >
                contact
              </a>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="center aligned ten wide column"
    >
      <div
        className="centered two column row"
      >
        <img
          alt="Not working today"
          className="logo"
          src="orca-logo-white.png"
        />
      </div>
    </div>
    <div
      className="three wide column"
    />
  </div>
</div>
`;
